.expeditor/scripts/finish_release/bump_rustfmt.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This script attempts to find the latest installable nightly rustfmt.
+# When found, it bumps the repo rustfmt version and submits a new PR
+# with the new version pin and any code base formatting changes from the
+# new version.
+
+# Assumptions:
+# 1. This script runs on a x86_64-unknown-linux-gnu target
+# 2. If rustfmt is installable on x86_64-unknown-linux-gnu, the other Tier1
+#    platforms are assumed to also have a working version.
+
+set -Eeuo pipefail
+
+# shellcheck source=.expeditor/scripts/verify/shared.sh
+source .expeditor/scripts/verify/shared.sh
+
+TODAY=$(date '+%Y-%m-%d')
+# pick out the date from a string like: nightly-2019-12-04
+CURRENT_RUSTFMT_DATE=$(awk -F- '{print $2"-"$3"-"$4}' RUSTFMT_VERSION)
+# convert to seconds for easier date comparison later
+CURRENT_RUSTFMT_DATE_S=$(date '+%s' --date "${CURRENT_RUSTFMT_DATE}")
+CHANNEL=nightly
+# a maximum iterator value for searching elligible nightlies
+MAX_ATTEMPTS=90
+
+# if set, this will be the next version
+export FOUND_VERSION
+
+install_rustup
+
+# set minimal profile so only core items + rustfmt are installed
+echo "--- :habicat: Setting minimal profile"
+rustup set profile minimal
+
+for ((i=0; i<MAX_ATTEMPTS; i++)); do
+  # begin searching nightly date versions using current date
+  # as a starting point
+  attempt_date=$(date '+%Y-%m-%d' --date "${TODAY} -${i} day")
+  # use the conversion to seconds for easy date comparison
+  attempt_date_s=$(date '+%s' --date "${attempt_date}")
+  attempt_version="${CHANNEL}-${attempt_date}"
+  echo "--- Trying Nightly Date Version: ${attempt_version}"
+  if rustup toolchain install "${attempt_version}" --component rustfmt; then
```

Comment:
My recollection is that we have to pick a nightly that offers the components we use (fmt, clippy, ?) for our  development platforms (Linux, Win, maybe Darwin?).  Maybe it's changed, but I believe that nightlies aren't guaranteed to ship with all components for all platforms.   

I believe this will potentially leave us in a place where our Windows linting in CI will start failing, if we were to happen to hit one of those releases.  I'll defer to @cm and @raskchanky on this one, as I may be operating on out of date information.

Reply:
I initially thought this as well, however I'm now convinced that the version from `RUSTFMT_VERSION` is used specifically only for running `rustfmt` via something similar to:

`cargo +$(< RUSTFMT_VERSION) fmt`

The nightly selected in the rustfmt case is only for that single component.

I definitely agree with your concerns about all the other components with respect to the version that is in `RUST_NIGHTLY_VERSION`. @raskchanky @christophermaier can you confirm?

Reply:
Yeah `RUSTFMT_VERSION` AFAIK is only used for verifying that the code has been properly formatted.  You can see that step [here](https://github.com/habitat-sh/habitat/blob/master/.expeditor/verify.pipeline.yml#L24) and follow the scripts through to verify.  As such, this only needs to work on linux, as that's the only place it ever runs.

Reply:
I've pushed https://github.com/habitat-sh/habitat/pull/7258/commits/cd84ed411ca62f354c6ecec4bdcbd9399d5c8781 that leverages `rustup set profile minimal` which allows us to use a nightly that doesn't necessarily have to have all the components like clippy when we just want to add `rustfmt`.